storage/operation/dbtest/helper.go

```diff
@@ -0,0 +1,60 @@
+package dbtest
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/operation/badgerimpl"
+	"github.com/onflow/flow-go/storage/operation/pebbleimpl"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// helper types and functions
+type WithWriter func(func(storage.Writer) error) error
+
+func RunWithStorages(t *testing.T, fn func(*testing.T, storage.Reader, WithWriter)) {
```

Comment:
The `RunWithStorages` creates two `t.Run`:
```
t.Run("BadgerStorage",
...
t.Run("PebbleStorage",
```

This allows us to define one test cases and run with two different database implementation.

Reply:
What I meant is that we could change the signature of `RunWithStorages` from 

```go
func RunWithStorages(t *testing.T, fn func(*testing.T, storage.Reader, WithWriter)) {
// ... body ...
}
```

to

```go
func RunWithStorages(fn func(*testing.T, storage.Reader, WithWriter)) func(t *testing.T) {
	return func(t *testing.T) {
           // ... exact same body ...
        }
}
```

so that the usage can change from 

```go
dbtest.RunWithStorages(t, func(t *testing.T, r storage.Reader, withWriter dbtest.WithWriter){
  // ... test body ...
}
```

to

```go
t.Run("My Test", dbtest.RunWithStorages(func(*testing.T, storage.Reader, WithWriter){
 // ... same test body ...
})
```

Its functionally almost the same, but the later approach allows you to nest the tests a little better.